cfgov/v1/models/snippets.py

```diff
@@ -48,3 +58,93 @@ def set_hash(sender, instance, **kwargs):
         heading = instance.heading.split(';;')[0]
 
     instance.heading = heading
+
+
+class ResourceTag(TaggedItemBase):
+    content_object = ParentalKey('v1.Resource', related_name='tagged_items')
+
+
+class TaggableSnippetManager(models.Manager):
+    def filter_by_tags(self, tags):
+        snippets = self.all()
+        for tag in tags or []:
+            snippets = snippets.filter(tags__name=tag)
+
+        return snippets
+
+
+@register_snippet
+class Resource(ClusterableModel):
+    title = models.CharField(max_length=255)
+    desc = RichTextField(verbose_name='Description', blank=True)
+
+    thumbnail = models.ForeignKey(
+        'v1.CFGOVImage',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+
+    related_file = models.ForeignKey(
+        'wagtaildocs.Document',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+
+    alternate_file = models.ForeignKey(
+        'wagtaildocs.Document',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+
+    link = models.URLField(
+        blank=True,
+        help_text='Example: URL to order a few copies of a printed piece.',
+        validators=[URLValidator]
+    )
+
+    alternate_link = models.URLField(
+        blank=True,
+        help_text='Example: a URL to for ordering bulk copies.',
+        validators=[URLValidator]
+    )
+
+    tags = TaggableManager(through=ResourceTag, blank=True)
+
+    objects = TaggableSnippetManager()
+
+    panels = [
+        FieldPanel('title'),
+        FieldPanel('desc'),
+        ImageChooserPanel('thumbnail'),
+        DocumentChooserPanel('related_file'),
+        DocumentChooserPanel('alternate_file'),
+        FieldPanel('link'),
+        FieldPanel('alternate_link'),
+        FieldPanel('tags'),
+    ]
+
+    # Makes fields available to the Actions chooser in a Snippet List module
+    snippet_list_field_choices = [
+        ('related_file', 'Related file'),
+        ('alternate_file', 'Alternate file'),
+        ('link', 'Link'),
+        ('alternate_link', 'Alternate link'),
+    ]
+
+    # Not needed now, but potentially useful if we ever want to render resource
+    # snippets individually.
+    # def template(self, is_single=False):
+    #    # return concatenation of subtype and kind of template
+    #    if is_single:
+    #       return self.subtype + '-single.html'
+    #    else:
+    #        return self.subtype + '-list-item.html'
```

Comment:
I can see it being something we'd want to do, but I can also see this approach not working. And good point about the lack of `subtype` definition. I'll just pull this chunk.